src/Draco.Compiler/Syntax/Lexer.cs

```diff
@@ -356,7 +357,31 @@ private IToken LexString()
             // Count the number of required delimiters
             for (var i = 0; i < mode.ExtendedDelims; ++i)
             {
-                if (this.Peek(offset + i) != '#') goto not_escape_sequence;
+                if (this.Peek(offset + i) != '#')
+                {
+                    offset--;
```

Comment:
We could also just not mutate offset when seeing the backslash initially?
That might be a better strategy to avoid the state juggling.

Reply:
But either way, great catch on the issue!